src/golang/lib/response/node.go

```diff
@@ -91,17 +126,55 @@ func NewNodesFromDBObjects(
 	operatorNodes []views.OperatorNode,
 	artifactNodes []views.ArtifactNode,
 ) *Nodes {
-	operators := make([]Operator, 0, len(operatorNodes))
-	artifacts := make([]Artifact, 0, len(artifactNodes))
-	for _, opNode := range operatorNodes {
-		operators = append(operators, *NewOperatorFromDBObject(&opNode))
+	return &Nodes{
+		Operators: functional_slice.Map(
+			operatorNodes,
+			func(node views.OperatorNode) Operator {
+				return *NewOperatorFromDBObject(&node)
+			},
+		),
+		Artifacts: functional_slice.Map(
+			artifactNodes,
+			func(node views.ArtifactNode) Artifact {
+				return *NewArtifactFromDBObject(&node)
+			},
+		),
 	}
+}
 
-	for _, artfNode := range artifactNodes {
-		artifacts = append(artifacts, *NewArtifactFromDBObject(&artfNode))
-	}
-	return &Nodes{
-		Operators: operators,
-		Artifacts: artifacts,
+type NodeResults struct {
+	Operators []OperatorResult `json:"operators"`
+	Artifacts []ArtifactResult `json:"artifacts"`
+}
+
+func NewNodeResultsFromDBObjects(
+	dbOperatorResults []models.OperatorResult,
+	dbArtifactResults []models.ArtifactResult,
+	contents map[string]string,
+) *NodeResults {
+	return &NodeResults{
+		Operators: functional_slice.Map(
+			dbOperatorResults,
+			func(result models.OperatorResult) OperatorResult {
+				return *NewOperatorResultFromDBObject(&result)
+			},
+		),
+		Artifacts: functional_slice.Map(
+			dbArtifactResults,
+			func(result models.ArtifactResult) ArtifactResult {
+				content, ok := contents[result.ContentPath]
+				var contentPtr *string
+				if ok {
+					contentPtr = &content
+				}
+
+				return *NewArtifactResultFromDBObject(&result, contentPtr)
+			},
+		),
 	}
 }
+
+type Content struct {
+	Name string `json:"name"`
+	Data []byte `json:"data"`
```

Comment:
I'm confused as to where this is used and what it represents. Can you add some explanation here?